iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ShuntCompensator.java

```diff
@@ -53,35 +40,32 @@
     ShuntCompensator setCurrentSectionCount(int currentSectionCount);
 
     /**
-     * Get the susceptance per section in S.
+     * Get the susceptance for the current section count.
+     * <p>
+     * Depends on the working variant.
+     * @see VariantManager
      */
-    double getbPerSection();
+    double getCurrentB();
 
     /**
-     * Set the susceptance per section in S.
-     *
-     * @param bPerSection the susceptance per section
-     * @return the shunt compensator to chain method calls.
+     * Get the conductance for the current section count.
+     * <p>
+     * Depends on the working variant.
+     * @see VariantManager
      */
-    ShuntCompensator setbPerSection(double bPerSection);
+    double getCurrentG();
 
     /**
-     * Get the susceptance for the maximum section count.
-     *
-     * @deprecated Use {@link #getbPerSection()} * {@link #getMaximumSectionCount()} instead.
+     * Get the model type of the shunt compensator (linear or non-linear)
      */
-    @Deprecated
-    default double getMaximumB() {
-        return getbPerSection() * getMaximumSectionCount();
-    }
+    ShuntCompensatorModelType getModelType();
 
     /**
-     * Get the susceptance for the current section counts.
-     * <p>
-     * Depends on the working variant.
-     * @see VariantManager
+     * Get the shunt model.
      */
-    double getCurrentB();
+    ShuntCompensatorModel getModel();
+
+    <M extends ShuntCompensatorModel> M getModel(Class<M> modelType);
```

Comment:
Should you add a default behaviour? Or do we decide that the API breaks

Reply:
For me, it would be clearer to break the API because I think keeping a mix between the old and new APIs would lead to more errors.

Reply:
Why do you create getModel() and getModelType? Couldn't the model be able to inform about its type?

Reply:
Yes but when calling it (from `ShuntCompensator`), we do not have this information, hence these two methods to retrieve it.

Reply:
Maybe  getModelType should be renamed getShuntType then?

Reply:
Isn't it a bit redundant to put "shuntType" in a method that only exists for shunts? (I put `getModelType()` because `getType()` already exists and returns which kind of connectable it is).